samples/AnomalyDetector/src/Generated/Models/AnomalyDetectionModel.Serialization.cs

```diff
@@ -39,7 +39,7 @@ void IJsonModel<AnomalyDetectionModel>.Write(Utf8JsonWriter writer, ModelReaderW
             if (Optional.IsDefined(ModelInfo))
             {
                 writer.WritePropertyName("modelInfo"u8);
-                writer.WriteObjectValue(ModelInfo);
+                ((IJsonModel<ModelInfo>)ModelInfo).Write(writer, options);
```

Comment:
there are three kinds of changes in this PR.
This PR only changes the code when the type of the property or the element of collection is a generated model (therefore it should implement the `IJsonModel<T>` interface)
This is the first kind: optional property.
We do not need the null check because `Optional.IsDefined` already did that